addon/components/fa-icon.js

```diff
@@ -75,26 +75,41 @@ const IconComponent = Component.extend({
     'viewBox',
     'safeStyle:style',
   ],
-  html: computed('children', function() {
-    const children = this.get('children')
-    let newHtml
-    if(!children){
+  html: computed('abstractIcon.children.[]', function() {
+    const abstractIcon = this.get('abstractIcon');
+    let newHtml;
+    if(!abstractIcon){
       newHtml = htmlSafe('')
     } else {
-      newHtml = htmlSafe(children.reduce((acc,cur) => {
+      newHtml = htmlSafe(abstractIcon.children.reduce((acc,cur) => {
         return `${acc}${toHtml(cur)}`
       },''))
     }
     return newHtml
   }),
-  safeStyle: computed('_frameworkStyle', function() {
-    const frameworkStyle = this.get('_frameworkStyle')
-    return frameworkStyle ? htmlSafe(`${this.get('_frameworkStyle')}`) : undefined
+  safeStyle: computed('attributes', function() {
+    const attributes = this.get('attributes');
+    const style = getWithDefault(attributes, 'style');
+    return style ? htmlSafe(`${style}`) : undefined;
   }),
-  didReceiveAttrs(){
-    this._super(...arguments)
-    /* eslint ember/no-attrs-in-components: 0 */
-    if('_frameworkStyle' in this.attrs) throw new Error('_frameworkStyle attribute is reserved for internal use and may not be set from a template')
+  abstractIcon: computed(
+    'prefix',
+    'class',
+    'icon',
+    'transform',
+    'mask',
+    'symbol',
+    'title',
+    'spin',
+    'pulse',
+    'fixedWidth',
+    'listItem',
+    'border',
+    'flip',
+    'size',
+    'rotation',
+    'pull',
+    function () {
     const iconLookup = normalizeIconArgs(this.get('prefix'), this.get('icon'))
```

Comment:
@jrjohnson Is this the `abstractIcon` block you're asking about in your previous comment?

> there is a whole block of code inside the new abstractIcon computed property that I didn't touch and don't really understand. Can you take a look at see if re-running this chunk over and over as dependent keys change is going to be a problem or if I missed a key somewhere?

If so, then yes, I think this would need to run when any of those dependent keys change. The "abstract icon" is basically an icon Abstract Syntax Tree, which may change very significantly as variations those dependent keys change. This is what can result an entirely different _shape_ for the SVG. So this `abstractIcon` code builds that AST, and then something like `toHtml()` renders the AST with appropriate DOM nodes. That's what's happening above in the `html` computed property.

Does that address the question? Or did I just talk _around_ it? Or something else?

Reply:
That's the right block and I think my changes work well inside that system I just wanted you to take a look now that we are no longer re-rendering anytime the attributes change, but instead depending on specific attributes as dependencies of the property.

Reply:
OK, well I haven't fine-tooth-combed it. But I did wide-tooth-comb it—a few times—and it looks good to me.